src/wallet/wallet.h

```diff
@@ -57,8 +58,13 @@ static const bool DEFAULT_SPEND_ZEROCONF_CHANGE = true;
 static const bool DEFAULT_SEND_FREE_TRANSACTIONS = false;
 //! -txconfirmtarget default
 static const unsigned int DEFAULT_TX_CONFIRM_TARGET = 2;
-//! Largest (in bytes) free transaction we're willing to create
+//! Largest (in bytes) free transaction we're willing to creat
+#ifdef BITCOIN_CASH
+//! We can allow a maximum sized free transaction in the Bitcoin Cash Network.
+static const unsigned int MAX_FREE_TRANSACTION_CREATE_SIZE = MAX_STANDARD_TX_SIZE;
```

Comment:
first change you made 
```
static const uint32_t DEFAULT_LIMITFREERELAY = DEFAULT_BLOCK_MAX_SIZE * 0.000015;
```
we limited the amount of zero fee / hiprio txns to be relayed to 15KB per minute, hence it would take ~66 minutes to get a 1MB transactions to be relayed between two nodes without any other zero fee txns passing by during that period. Am I correct?

Reply:
@sickpig not sure what you mean about ~66 mins?.  The only change is that the limitfreerelay value will rise with blocksize, but there is no change in current behavior at the 1MB block level.  Basically there is a 10minute window which decays..so every 10 minutes you can send/route 150KB of free txns in that window. (or 15KB per minute), but whether they get mined or not is up to the miners.

Reply:
`MAX_FREE_TRANSACTION_CREATE_SIZE = 1MB`
implies that I could create a free transaction as big as 1MB, but as you said the net (by default) can only send/route 150KB of free every 10 minutes. 

I didn't looked at the code but suppose that the "counter" that keep track of free txns is not restarted every new  block.

That means that I can create a 1MB free transaction and that it would be succesfully relayed to another node only if no other free txns is sent/routed for at 6/7 blocks (1MB / 15 KB/min) = 66 minutes.

Am I wrong?

Reply:
@sickpig The biggest free txn is only MAX_STANDARD_TX_SIZE or 100KB...i think you didn't see that and thought it was max block size.

Reply:
I mean shouldn't we anchor `MAX_FREE_TRANSACTION_CREATE_SIZE` to `DEFAULT_BLOCK_MAX_SIZE` like we did for `DEFAULT_LIMITFREERELAY`  ?

Reply:
@sickpig also it' a little confusing because the decay window is actually every 10 minutes, not every minute...in the past they should have really made limit free relay 150  instead of 15 (and they they multiply by 10 at some point), but I hesitate to tidy that up because it involves a little refactoring.

Reply:
@sickpig oh i see, about the anchoring...no i don't think so ...the max txn is 100KB and is a routing policy across the network...we can't just arbitrarily up that figure otherwise our txns won't get routed. And really there is very little use case for txns greater than 100KB.  I think we had this during the SigOps round of discussions.

Reply:
ouch you are right, I read 1MB when it was 100KB. Sorry for the noise.